redact/v3/redact_requests.py

```diff
@@ -193,29 +199,41 @@ def _stream_output_to_file(
                 "wb", dir=str(file.parent), delete=False
             )
             finished = False
+            target_file = Path(temp_file.name)
             try:
                 with temp_file:
                     for chunk in response.iter_bytes():
                         temp_file.write(chunk)
                 finished = True
             finally:
                 if finished:
-                    file_name = Path(retrieve_file_name(headers=response.headers))
-                    anonymized_path = Path(file.parent).joinpath(
-                        f"{file.stem}{file_name.suffix}"
```

Comment:
this behavior is misleading, because of this side effect.
We save the file not by the provided path, but the one that resembles the original file name (specifically suffix),
which might be different from method call argument suffix.